erpnext/stock/get_item_details.py

```diff
@@ -624,7 +624,11 @@ def get_item_price(args, item_code, ignore_party=False):
 		elif args.get("supplier"):
 			conditions += " and supplier=%(supplier)s"
 		else:
-			conditions += " and (customer is null or customer = '') and (supplier is null or supplier = '')"
+			conditions += """and case when ifnull(selling, 0) = 1 then 
+								(customer is null or customer = '')	
+							 else
+								(supplier is null or supplier = '')
+							 end"""
```

Comment:
```suggestion
			conditions += " and (customer is null or customer = '') or (supplier is null or supplier = '')"
```

Reply:
The current fix fails where Buying and Selling are both enabled.
Assume an Item Price is created for buying and selling both. For a buying transaction if Supplier is not set it should fetch it anyway. But, assume the Customer is set because we want it to fetch for a particular customer in selling, it fails. 
We dont know if the transaction is buying or selling in this area. So, checking for buying or selling Item Price is also incorrect. 'OR' should do the trick, if either aren't set it will fetch it (because it tried checking for customer/supplier once already).